### ISSUE TITLE
`VisualScriptFunctionCall.set_basic_type` not validating input which later cause crash

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#53456]
### Godot version

3.4.beta.custom_build. 01ae4888e

### System information

Ubuntu 21.04 - Nvidia GTX 970, Gnome shell 3.38 X11

### Issue description

When executing
```
	var temp_variable12807 = VisualScriptFunctionCall.new()
	temp_variable12807.set_call_mode(3)
	temp_variable12807.set_basic_type(-32)
	temp_variable12807._get_default_input_values()
```

Godot crashes with backtrace
```
modules/visual_script/visual_script_func_nodes.cpp:133:68: runtime error: load of value 4294967264, which is not a valid value for type 'Type'

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v3.4.beta.custom_build (e848da3a5d535f9df56613b7a2ad2371327b511a)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] godots() [0x17d812a] (/mnt/Miecz/godot3.2/platform/x11/crash_handler_x11.cpp:56)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x41040) [0x7f931ec11040] (??:0)
[3] Map<StringName, _VariantCall::FuncData, Comparator<StringName>, DefaultAllocator>::find(StringName const&) const (/mnt/Miecz/godot3.2/./core/map.h:509)
[4] Variant::get_method_argument_types(Variant::Type, StringName const&) (/mnt/Miecz/godot3.2/core/variant_call.cpp:1395)
[5] VisualScriptFunctionCall::get_input_value_port_count() const (/mnt/Miecz/godot3.2/modules/visual_script/visual_script_func_nodes.cpp:134)
[6] VisualScriptNode::_get_default_input_values() const (/mnt/Miecz/godot3.2/modules/visual_script/visual_script.cpp:99)
[7] MethodBind0RC<Array>::call(Object*, Variant const**, int, Variant::CallError&) (/mnt/Miecz/godot3.2/./core/method_bind.gen.inc:593 (discriminator 4))
[8] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/mnt/Miecz/godot3.2/core/object.cpp:918 (discriminator 1))
[9] Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) (/mnt/Miecz/godot3.2/core/variant_call.cpp:1175 (discriminator 1))
[10] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/mnt/Miecz/godot3.2/modules/gdscript/gdscript_function.cpp:1049)
[11] GDScriptInstance::_ml_call_reversed(GDScript*, StringName const&, Variant const**, int) (/mnt/Miecz/godot3.2/modules/gdscript/gdscript.cpp:1199)
[12] GDScriptInstance::call_multilevel_reversed(StringName const&, Variant const**, int) (/mnt/Miecz/godot3.2/modules/gdscript/gdscript.cpp:1207)
[13] Node::_notification(int) (/mnt/Miecz/godot3.2/scene/main/node.cpp:149)
[14] Node::_notificationv(int, bool) (/mnt/Miecz/godot3.2/./scene/main/node.h:45 (discriminator 14))
[15] CanvasItem::_notificationv(int, bool) (/mnt/Miecz/godot3.2/./scene/2d/canvas_item.h:163 (discriminator 3))
[16] Node2D::_notificationv(int, bool) (/mnt/Miecz/godot3.2/./scene/2d/node_2d.h:37 (discriminator 3))
[17] Object::notification(int, bool) (/mnt/Miecz/godot3.2/core/object.cpp:929)
[18] Node::_propagate_ready() (/mnt/Miecz/godot3.2/scene/main/node.cpp:188)
[19] Node::_propagate_ready() (/mnt/Miecz/godot3.2/scene/main/node.cpp:178 (discriminator 2))
[20] Node::_set_tree(SceneTree*) (/mnt/Miecz/godot3.2/scene/main/node.cpp:2560)
[21] SceneTree::init() (/mnt/Miecz/godot3.2/scene/main/scene_tree.cpp:465)
[22] OS_X11::run() (/mnt/Miecz/godot3.2/platform/x11/os_x11.cpp:3636)
[23] godots(main+0x326) [0x17cedac] (/mnt/Miecz/godot3.2/platform/x11/godot_x11.cpp:57)
[24] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xd5) [0x7f931ebf8565] (??:0)
[25] godots(_start+0x2e) [0x17ce9ce] (??:?)

```

Code is here and p_type should be between 0(Variant::Nil) and Variant::Max
https://github.com/godotengine/godot/blob/a7ba22763139b29d825e0f4ef3e718533da069ea/modules/visual_script/visual_script_func_nodes.cpp#L299-L307

This example was found by fuzzer, so I don't recommend trying understand in what situation such code could be used in real project, because such situation probably doesn't exists.

### Steps to reproduce

Above

**Comment by Nimmda** [comment]
Hi, would this be sufficient? Does the user need feedback in godot?

```cpp
void VisualScriptFunctionCall::set_basic_type(Variant::Type p_type) {
	if (basic_type == p_type) {
		return;
	}

	if (p_type < Variant::NIL || p_type > Variant::VARIANT_MAX) {
		return;
	}

	basic_type = p_type;

	_change_notify();
	ports_changed_notify();
}
````

**Comment by akien-mga** [comment]
@Nimmda This should typically throw an error like done in #53464. But there's already a PR so this doesn't need a new fix, only reviews from core contributors.

**Comment by akien-mga** [comment]
Fixed by #53464.
